chapter_2/exercise_2_02/loop.c

```diff
@@ -22,14 +22,17 @@ int main(void)
     if (i >= (MAXLINE - 1))
     {
       loop = 0;
+      break;
     }
     else if (c == '\n')
     {
       loop = 0;
+      break;
     }
     else if (c == EOF)
     {
       loop = 0;
+      break;
     }
```

Comment:
@SamiNasry This solution works well. We can improve the code by merging all the conditions into one condition with logical OR. Then, we can have only one set for `loop = 0` and only one break.

Reply:
So, let's convert this: 
```c
if (i >= (MAXLINE - 1))
{
  loop = 0;
  break;
}
else if (c == '\n')
{
  loop = 0;
  break;
}
else if (c == EOF)
{
  loop = 0;
  break;
}
```

into something like this:
```c
if (i >= (MAXLINE - 1) || c == '\n' || c == EOF)
{
  loop = 0;
}
```

I find this approach more compact and a bit easier to read. If you have a different opinion, please let me know.

@illustrofia mentions this is also an approach worth considering. I like it, but we're doing the same operation many times, which is a bit counterproductive. Also, having the same thing happening multiple times in multiple places will facilitate a pattern that might end up in divergence over time, especially in an environment where many devs are involved.
```c
if (i >= (MAXLINE - 1))
{
  loop = 0;
}

if (c == '\n')
{
  loop = 0;
}

if (c == EOF)
{
  loop = 0;
}
```

Reply:
@SamiNasry I agree with you, except for the break, it doesn't make logical sense to me. I'll investigate it a bit and come back to you.

Reply:
Yes, try to get the original getline function and move the LIMIT to 10 and in your code move the MAXLINE to 10 and try to type strings with 10 charchters or more you will find the behavior is not the same for both codes. Without the break the code adds an extra charachter.

Reply:
You are right we have different behaviors. That's an interesting problem I will look into it further.
This is how it acts for me with the break
![image](https://github.com/user-attachments/assets/cfdeefb7-4f9f-4a7d-b6f6-644c48c30ae5)

This is how it acts without the break
![image](https://github.com/user-attachments/assets/140bf504-bbcb-4290-af23-b4548e40c499)


Anyways I will try to look into why it acts like this in different machines. For now let's just use the more compact solution. Thanks for your time.

Reply:
Can you please show me how does the original getline function behave for the same input in your machine ?

Reply:
This is the code of the original getline() funtion 
```C
#include <stdio.h>

int getline(char s[], int lim);

int getline(char s[], int lim)
{
	int c,i;
	for (i = 0; i < lim -1 && ((c = getchar()) != EOF) && c != '\n' ; ++i)
	{
		s[i] = c;

	}
	if (c == '\n') {
		s[i] = c;
		++i;
	}
	s[i] = '\0';
	return i;
}


int main()
{
	int lim = 10;
	char s[lim];

	
	getline(s, lim);
	printf("%s" , s);
}

```

Reply:
<img width="528" alt="image" src="https://github.com/user-attachments/assets/0ef088ce-eb50-4049-9abc-fc8cc2bfa9b7">

@SamiNasry Here is how it behaves for me. I had to change the name, since the `clang` compiler doesn't like me to reuse the getline name since it's already part of the stdio lib.